src/main/java/at/hannibal2/skyhanni/utils/OtherModsSettings.kt

```diff
@@ -1,25 +1,108 @@
+                runCatching {
+                    clazz.getField(name).apply { isAccessible = true }
+                }.getOrElse {
+                    clazz.getDeclaredField(name).apply { isAccessible = true }
+                }
+            }
+
+        fun clearClassCache() { classCache.clear() }
+        fun clearFieldCache() { fieldCache.clear() }
+        fun clearCaches() {
+            clearClassCache()
+            clearFieldCache()
+        }
     }
 
-    fun getBoolean(optionPath: String): Boolean = getOption(optionPath)?.get(null) as? Boolean ?: false
+    fun isLoaded(): Boolean = loadClass(modConfigPath) != null
+
+    private val chainCache = ConcurrentHashMap<Pair<String, String>, List<Field>>()
 
-    fun setBoolean(optionPath: String, value: Boolean) {
-        getOption(optionPath)?.set(null, value)
+    fun getBoolean(path: String): Boolean? = getNestedValue<Boolean>(path)
+    fun setBoolean(path: String, value: Any) = setNestedValue(path, value)
+    fun getConfigValue(path: String): Any? = getNestedValue<Any>(path)
+    fun setConfigValue(path: String, value: Any) = setNestedValue(path, value)
+
+    private inline fun <reified T> getNestedValue(path: String): T? {
+        loadClass(modConfigPath) ?: return null
+        val key = modConfigPath to path
+        val chain = chainCache.computeIfAbsent(key) { buildFieldChain(path) }
+        if (chain.isEmpty()) return null
+
+        val (rootInstance, firstField) = getOptionPair(chain[0].name) ?: return null
+        var current: Any? = runCatching { firstField.get(rootInstance) }.getOrNull()
+
+        for (f in chain.drop(1)) {
+            current = current?.let { runCatching { f.get(it) }.getOrNull() } ?: return null
+        }
+        return current as? T
     }
 
-    private fun getOption(optionPath: String): Field? =
-        try {
-            Class.forName(modConfigPath).getField(optionPath)
-        } catch (e: Throwable) {
-            null
+    private fun setNestedValue(path: String, value: Any) {
+        loadClass(modConfigPath) ?: return
+        val key = modConfigPath to path
+        val chain = chainCache.computeIfAbsent(key) { buildFieldChain(path) }
+        if (chain.size < 2) return
+
+        val (rootInstance, firstField) = getOptionPair(chain[0].name) ?: return
+        var current: Any? = runCatching { firstField.get(rootInstance) }.getOrNull()
+
+        for (f in chain.drop(1).dropLast(1)) {
+            current = current?.let { runCatching { f.get(it) }.getOrNull() } ?: return
+        }
+        runCatching { chain.last().set(current, value) }
+    }
+
+    private fun buildFieldChain(path: String): List<Field> {
+        val segments = path.split('.')
+        val rootPair = getOptionPair(segments.first()) ?: return emptyList()
+        val fields = mutableListOf(rootPair.second)
+        var currentClass = rootPair.second.type
+        for (seg in segments.drop(1)) {
```

Comment:
please dont use abbreviations like this that arent very common in the kotlin/hypixel sb scene